Source/Engine/AudioEngine.swift

```diff
@@ -201,4 +203,52 @@ class AudioEngine: AudioEngineProtocol {
     func invalidate() {
         
     }
+
+    private func scaledPower(power: Float) -> Float {
+        guard power.isFinite else { return 0.0 }
+        let minDb: Float = -80.0
+        if power < minDb {
+            return 0.0
+        } else if power >= 1.0 {
+            return 1.0
+        } else {
+            return (abs(minDb) - abs(power)) / abs(minDb)
+        }
+    }
+
+    private func connectVolumeTap() {
+        let format = engine.mainMixerNode.outputFormat(forBus: 0)
+        engine.mainMixerNode.installTap(onBus: 0, bufferSize: 1024, format: format) { [weak self] buffer, when in
+            guard let self = self, SAPlayer.shared.skipSilences else { return }
+            guard let channelData = buffer.floatChannelData else {
+                return
+            }
+
+            let channelDataValue = channelData.pointee
+            let channelDataValueArray = stride(from: 0,
+                                               to: Int(buffer.frameLength),
+                                               by: buffer.stride).map { channelDataValue[$0] }
+
+            let rms = sqrt(channelDataValueArray.map { $0 * $0 }.reduce(0, +) / Float(buffer.frameLength))
+
+            let avgPower = 20 * log10(rms)
+
+            let meterLevel = self.scaledPower(power: avgPower)
+            if meterLevel < 0.6 {
+                self.changeRate(to: 1.5)
```

Comment:
I don't want to use hardcoded 1.5 for speeding up to skip silences as they might have the audio rate set to 4 or something by default. 

There's 2 options here:
1. We change this to add to the current rate by some appropriate factor. I'm think this rate increase for skipping silences should be less at higher rates so we'd need to determine some sort of ratio.
2. We assume that at rates above some number, like 3, the skip silences feature wouldn't be noticeable/helpful and then we could just restrict the rate change to a lower range of original rates. This makes it easy to just add some constant to the current rate.

Reply:
That's a really good point that I hadn't considered. I like both suggestions and think a mix of the two would be best? Applying some appropriate factor definitely seems the right way to go and if the rate increases beyond to a point where the skip would be negligible it just switches the tap off?